sympy/core/power.py

```diff
@@ -1694,7 +1694,7 @@ def mul(d1, d2):
             res += terms[e1]*inco*x**(ex)
 
         for i in (1, 2, 3):
-            if (res - self).subs(x, i) is not S.Zero:
+            if (res - self).cancel().subs(x, i) is not S.Zero:
```

Comment:
There may be other possibilities, e.g `expand` or `_mexpand`. We should choose one that is most efficient in general. (Many tests may be needed though.)

Reply:
This is supposed to be a quick test.

Reply:
An order term should be added whenever the expansion of `base**exp` is truncated. That will not happen when `exp` is a positive integer and sufficiently small in comparison with `n`, In that case, it seems that `_mexpand` is exactly what is needed to transform `base**exp`. Hence I would consider something like this (instead of the `subs` trick):
```
        if not (exp.is_integer and exp.is_positive and exp*d < n and
                res == _mexpand(self)):
            res += O(x**n, x)
```
(The purpose of `exp*d < n` is to avoid expanding something like `base**10000`.)

Reply:
Would this check take care of the `base**10000` case?https://github.com/sympy/sympy/blob/62572863590b1c3363d6268f75aea6e83e385728/sympy/core/power.py#L1623-L1625

Reply:
It is possible. Maybe you could run tests.

Reply:
However, `m` may be 0 or negative.

Reply:
The maxpow condition was indeed insufficient. An example is https://github.com/sympy/sympy/blob/62572863590b1c3363d6268f75aea6e83e385728/sympy/series/tests/test_limits.py#L725-L726 After this change, this example runs almost instantaneously compared to taking a couple of seconds earlier